### DISCUSSION TITLE
Interacting with the comment threads programmatically

### DISCUSSION THREAD

**Discussion by aayush420** [microsoft/vscode-discussions#336] [Extension Development QnA] [answered]
Is there a way to get the CommentThread given the range and the document? It would be useful to add comments programmatically, rather than using the UI element on the line number column. For example, let's say that I have a codelens button that I want to use to add a certain comment on the commentthread at the range/position where the codelens button is showing up.

**Comment by isidorn** [comment, marked as answer]
There is the https://code.visualstudio.com/api/references/vscode-api#comments API, please try it out.
I am not sure the exact usage, but you can inspect this API and figure this out on your own.
